src/layouts/Workspace/Workspace.tsx

```diff
@@ -19,123 +22,75 @@ import {
 
 import { ProtectedRouteWithProps } from "routing/ProtectedRouteWithProps"
 
-import { isDirData } from "types/utils"
+import { DirectoryData } from "types/directory"
 
-import {
-  Section,
-  SectionHeader,
-  SectionCaption,
-  CreateButton,
-} from "../components"
-import { SiteViewLayout } from "../layouts"
+import { ContentGridLayout, SiteViewLayout } from "../layouts"
 
 import {
-  ContactCard,
-  PageCard,
-  NavigationCard,
-  FolderCard,
-  HomepageCard,
-} from "./components"
-
-const CONTACT_US_TEMPLATE_LAYOUT = "contact_us"
+  MainPages,
+  UngroupedPages,
+  WorkspaceFolders,
+} from "./components/WorkspacePagesAndFoldersComponents"
 
 const WorkspacePage = (): JSX.Element => {
   const {
     params: { siteName },
   } = useRouteMatch<{ siteName: string }>()
 
-  const { setRedirectToPage } = useRedirectHook()
   const { url } = useRouteMatch()
-  const { data: _dirsData, isLoading: isDirLoading } = useGetFolders({
+  const {
+    isLoading: isDirLoading,
+    data: _dirsAndPagesData,
+  } = useGetFoldersAndPages({
     siteName,
   })
-  const { data: _pagesData, isLoading: isPagesLoading } = useGetWorkspacePages(
-    siteName
-  )
-  const { data: contactUsPage, isLoading: isContactUsLoading } = useGetPageHook(
-    {
-      siteName,
-      fileName: "contact-us.md",
-    }
-  )
+  const { data: _pagesData } = useGetWorkspacePages(siteName)
 
-  const hasContactUsCard =
-    contactUsPage?.content?.frontMatter?.layout === CONTACT_US_TEMPLATE_LAYOUT
-
-  const dirsData = _dirsData?.filter(isDirData) || []
   const pagesData = _pagesData || []
+  const dirsData = (_dirsAndPagesData || []).filter((data) => {
+    return data.type === "dir" // only get directories and not pages
+  }) as DirectoryData[]
```

Comment:
we should not need the cast here - there should be a `isDirData` typeguard in `types/utils`. using that should prevent the type casting (through the return type of the function being asserted as `x is y`, where y is the type)

Reply:
cce8135